adapters/characters_repository.go

```diff
@@ -0,0 +1,67 @@
+package adapters
+
+import (
+	"bootcamp/domain/model"
+	"bootcamp/infra/datastore"
+	"bootcamp/usecase/interfaces"
+	"fmt"
+	"strconv"
+	"strings"
+)
+
+type fileStoreCharacterRepository struct {
+	characters    []*model.Character
+	charactersMap map[int]*model.Character
+}
+
+func NewCharacterRepository() interfaces.CharacterRepository {
+	return &fileStoreCharacterRepository{}
+}
+
+func (repository *fileStoreCharacterRepository) load() error {
```

Comment:
This approach is nice. Just in case you want to consider it, [here](https://github.com/gocarina/gocsv)'s a CSV reader that I've found easier to use than the native one.